src/pinnable_mapped_file.cpp

```diff
@@ -216,7 +300,7 @@ void pinnable_mapped_file::setup_non_file_mapping() {
    round_up_mmaped_size(_2mb);
    _non_file_mapped_mapping = mmap(NULL, _non_file_mapped_mapping_size, PROT_READ|PROT_WRITE, common_map_opts, VM_FLAGS_SUPERPAGE_SIZE_2MB, 0);
    if(_non_file_mapped_mapping != MAP_FAILED) {
-      std::cerr << "CHAINBASE: Database \"" << _database_name << "\" using 2MB pages" << std::endl;
+      std::cerr << "CHAINBASE: Database \"" << _database_name << "\" using 2MB pages" << '\n';
```

Comment:
I get a warning from clang-tidy that it is a performance issue. I believe [this](https://www.geeksforgeeks.org/avoid-using-stdendl/) is the reason.

Reply:
But for log messages you typically do want the flush at the end of a message. Otherwise you have no guarantee how promptly the output will be visible to the user (or recorded in to some log ingest system). In the case of the `% complete` messages, without the flush none of these may be output until after 100% is already reached.

This can be seen with something like
```c++
#include <iostream>
#include <thread>
int main() {
	using namespace std::chrono_literals;
	std::cout << "Get Ready!" << std::endl;
	for(unsigned i = 0; i < 60; ++i) {
	        std::cout << i << '\n';
	        std::this_thread::sleep_for(1s);
	}
	return 0;
}
```
Then run something like
```
./o > output.txt & tail -f output.txt
```
and you will likely not see any output beyond `Get Ready` for an extended period of time (for me, until the application exits). Using `endl` instead shows each number counting up.

So for a user who sends stderr to a log file they won't see our `% complete` messages promptly.

Reply:
I'd rather not add it back. [clang-tidy](https://clang.llvm.org/extra/clang-tidy/checks/performance/avoid-endl.html) warns about it. And personally I don't find it more readable.

Reply:
> `cerr` is non-buffered

good point, I had forgot there was a difference there. So this change is really just ultimately a style change, and the clang-tidy commentary is not applicable

Reply:
> But for log messages you typically do want the flush at the end of a message.

OK, I'll add the flush back. Would you prefer I revert to what it was before, or I use `std::cout << "Hello\n" << std::flush;` as mentioned [here](https://clang.llvm.org/extra/clang-tidy/checks/performance/avoid-endl.html).

Reply:
> But for log messages you typically do want the flush at the end of a message.

Like @heifner pointed out, since it's `cerr` it's not buffered by default anyways, so that comment from me isn't accurate in this case.

Reply:
Personally, I'm fine either way (leaving as is or reverting or using flush or ..)

Reply:
@spoonincode at this point I'm happy to make any change (or none) that are necessary for your approval today. Just let me know.